examples/asset/temp_asset.rs

```diff
@@ -0,0 +1,183 @@
+//! This example shows how to use the temporary asset source, `temp://`.
+//! First, a [`TextAsset`] is created in-memory, then saved into the temporary asset source.
+//! Once the save operation is completed, we load the asset just like any other file, and display its contents!
+
+use bevy::{
+    asset::{
+        saver::{AssetSaver, ErasedAssetSaver},
+        AssetPath, ErasedLoadedAsset, LoadedAsset,
+    },
+    prelude::*,
+    tasks::{block_on, IoTaskPool, Task},
+};
+
+use futures_lite::future;
+use text_asset::{TextAsset, TextLoader, TextSaver};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .init_asset::<TextAsset>()
+        .register_asset_loader(TextLoader)
+        .add_systems(Startup, (save_temp_asset, setup_ui))
+        .add_systems(Update, (wait_until_temp_saved, display_text))
+        .run();
+}
+
+/// Attempt to save an asset to the temporary asset source.
+fn save_temp_asset(assets: Res<AssetServer>, mut commands: Commands) {
+    // This is the asset we will attempt to save.
+    let my_text_asset = TextAsset("Hello World!".to_owned());
+
+    // To ensure the `Task` can outlive this function, we must provide owned versions
+    // of the `AssetServer` and our desired path.
+    let path = AssetPath::from("temp://message.txt").into_owned();
+    let server = assets.clone();
+
+    let task = IoTaskPool::get().spawn(async move {
+        save_asset(my_text_asset, path, server, TextSaver)
+            .await
+            .unwrap();
+    });
```

Comment:
I feel like this should be a method on `AssetServer` just like `load`. `load` already uses a pretty similar process as this (take a clone of the asset server, spawn an `IoTaskPool`, etc.), and ergonomically this just looks bad, especially when you peak inside `save_asset`.

Reply:
I want this, but would be more appropriate in its own PR for the `AssetServer`.